newIDE/app/src/MainFrame/index.js

```diff
@@ -1793,6 +1840,7 @@ class MainFrame extends React.Component<Props, State> {
                 label={editorTab.name}
                 key={editorTab.key}
                 active={isCurrentTab}
+                onSetPreview={() => this._onSetPreview(editorTab)}
```

Comment:
onSetPreview is something that is not really related to tabs. It's just a way to say "hey, please start the game with this scene". So give the responsibility to the editor to pass the scene name to `onSetPreview`.